### PR TITLE
Feature/aa 253 add endpoint to get notification options for agent

### PR DESCRIPTION
**Pull request by AlexeyShok**
Added endpoint for get materialized notification options for agent

### CODE REVIEW

File: Api/NotificationCenter/Services/NotificationOptionsService.cs [hunk 1]
```diff
@@ -28,6 +32,26 @@ public async Task<Result<SlimNotificationOptions>> GetNotificationOptions(int us
         }
 
 
+        public async Task<Dictionary<NotificationTypes, SlimNotificationOptions>> Get(SlimAgentContext agent)
+        {
+            var agentOptions = await _context.NotificationOptions
+                .Where(no => no.AgencyId == agent.AgencyId && no.UserId == agent.AgentId && no.UserType == ApiCallerTypes.Agent)
```
Comment (oleg1std) [inline Api/NotificationCenter/Services/NotificationOptionsService.cs:38]: maybe it is better to use _o_ instead of _no_

Comment (AlexeyShok) [reply Api/NotificationCenter/Services/NotificationOptionsService.cs:38]: NotificationOption => no

**Review by hasmikharutyunyan — APPROVED** [review]

File: Api/Controllers/AgentControllers/NotificationController.cs [hunk 2]
```diff
@@ -40,7 +43,23 @@ public async Task<IActionResult> GetNotifications([FromQuery] int skip = 0, [Fro
...
+        /// </summary>
+        /// <returns>List of notification settings</returns>
+        [HttpGet("settings")]
+        [ProducesResponseType(typeof(Dictionary<NotificationTypes, NotificationSettings>), (int)HttpStatusCode.OK)]
+        [AgentRequired]
+        public async Task<IActionResult> GetNotificationOptions()
+        {
+            var agent = await _agentContextService.GetAgent();
+
+            return Ok(await _notificationOptionsService.Get(new SlimAgentContext(agent.AgentId, agent.AgencyId)));
```
Comment (kirillta) [inline Api/Controllers/AgentControllers/NotificationController.cs:57]: This one is a little bit complicated, could be splatted to two lines

File: Api/NotificationCenter/Services/NotificationOptionsService.cs [hunk 3]
```diff
@@ -79,6 +104,35 @@ private Task<NotificationOptions> GetOptions(int userId, ApiCallerTypes userType
                 .SingleOrDefaultAsync(o => o.UserId == userId && o.UserType == userType && o.AgencyId == agencyId && o.Type == notificationType);
 
 
+        private static Dictionary<NotificationTypes, NotificationSettings> GetMaterializedOptions(List<NotificationOptions> userOptions)
```
Comment (kirillta) [inline Api/NotificationCenter/Services/NotificationOptionsService.cs:107]: Why blank lines place after every code line in this method?

File: Api/NotificationCenter/Services/NotificationOptionsService.cs [hunk 4]
```diff
@@ -79,6 +104,35 @@ private Task<NotificationOptions> GetOptions(int userId, ApiCallerTypes userType
...
+            foreach (var option in defaultOptions)
+            {
+                var userOption = userOptions.SingleOrDefault(no => no.Type == option.Key);
+
+                var enabledProtocols = new Dictionary<ProtocolTypes, bool>();
+
+                if (option.Value.EnabledProtocols.HasFlag(ProtocolTypes.Email))
+                {
+                    enabledProtocols.Add(ProtocolTypes.Email, userOption is null || option.Value.IsMandatory || userOption.EnabledProtocols.HasFlag(ProtocolTypes.Email));
+                }
```
Comment (kirillta) [inline Api/NotificationCenter/Services/NotificationOptionsService.cs:122]: Redundant braces

File: Api/Controllers/AgentControllers/NotificationController.cs (hunk 2 above)
Comment (AlexeyShok) [reply Api/Controllers/AgentControllers/NotificationController.cs:57]: Fixed

File: Api/NotificationCenter/Services/NotificationOptionsService.cs (hunk 3 above)
Comment (AlexeyShok) [reply Api/NotificationCenter/Services/NotificationOptionsService.cs:107]: Removed blank lines

File: Api/NotificationCenter/Services/NotificationOptionsService.cs (hunk 4 above)
Comment (AlexeyShok) [reply Api/NotificationCenter/Services/NotificationOptionsService.cs:122]: Didn't see any redundant braces, maybe I fixed it earlier?

**Review by kirillta — APPROVED** [review]

**Pull request merged**